Tests/EndToEnd.tests.ts

```diff
@@ -968,4 +976,94 @@ describe("EndToEnd", () => {
             }
         });
     });
+
+    describe("Heartbeat metrics for VM", () => {
+        var sandbox: sinon.SinonSandbox;
+
+        beforeEach(() => {
+            sandbox = sinon.sandbox.create();
+        });
+
+        afterEach(() => {
+            sandbox.restore();
+        });
+
+        it("should collect correct VM information from JSON response", (done) => {
+            // set up stub
+            const vmDataJSON = `{
+                "vmId": "1",
+                "subscriptionId": "2",
+                "osType": "Windows_NT"
+            }`;
+            var stub: sinon.SinonStub = sandbox.stub(http, "request", (options: any, callback: any) => {
+                var req = new fakeRequest(false, "http://169.254.169.254", vmDataJSON);
```

Comment:
make sure I return a JSON object as response - then metrics properties populate correct vm info from the respond data.